cyberpandas/ip_array.py

```diff
@@ -305,15 +305,23 @@ def isin(self, other):
             except ValueError:
                 networks.append(ipaddress.IPv6Network(net))
 
-        # TODO: perf
-        pyips = self.to_pyipaddress()
         mask = np.zeros(len(self), dtype='bool')
         for network in networks:
-            for i, ip in enumerate(pyips):
-                if ip in network:
-                    mask[i] = True
+            mask |= self._isin_network(network)
         return mask
 
+    def _isin_network(self, other):
+        # type: (Union[ipaddress.IPv4Network,ipaddress.IPv6Network]) -> ndarray
+        """Check whether an array of addresses is contained in a network."""
+        # A network is bounded below by 'network_address' and
+        # above by 'broadcast_address'.
+        # IPArray handles comparisons between arrays of addresses, and NumPy
+        # handles broadcasting.
+        net_lo = type(self)([other.network_address])
```

Comment:
Will this fail on attempt to compare IP4 and IP6?

Reply:
Once we're in an IPArray, we don't care about IPv4 vs. IPv6. Everything is 128 bits.